### PR TITLE
[TT-10386] Temporal Storage Redis CI

### PR DESCRIPTION
**Pull request by mativm02**
## Description

Adding a new CI pipeline to run tests in `temporal` pkg.
## Related Issue

https://tyktech.atlassian.net/browse/TT-10386?atlOrigin=eyJpIjoiMzU5ZTc5NTFmYzFkNGNlNjk0NGU5YmUyYzhjYzU0ZGMiLCJwIjoiaiJ9
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)
- [ ] Documentation updates or improvements.

## Checklist

- [ ] I have [reviewed the guidelines](./CONTRIBUTING.md) for contributing to this repository.
- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If PRing from your fork, don't come from your `master`!
- [ ] Make sure you are making a pull request against our **`master` branch** (left side). Also, it would be best if you started *your change* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
     - [ ] I have manually updated the README(s)/documentation accordingly.
     - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] Modules and vendor dependencies have been updated; run `go mod tidy && go mod vendor`
- [ ] When updating library version must provide reason/explanation for this update.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] Check your code additions will not fail linting checks:
    - [ ] `gofmt -s -w .`
    - [ ] `go vet ./...`

### CODE REVIEW

**Review by sredxny — APPROVED** [review]

File: bin/temporal-storage-tests.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
```
Comment (tbuchaillot) [inline bin/temporal-storage-tests.sh:1]: this is pretty much duplicated of `ci-test.sh`, let's make it more generic.

**Review by tbuchaillot — CHANGES_REQUESTED** [review]
As we talked, let's try a more generic taskfile approach!

File: .github/workflows/ci-tests.yml [hunk 2]
```diff
@@ -73,16 +82,29 @@ jobs:
...
-      - name: Run tests
+          mongodb-version: "${{ matrix.version }}"
+
+      - if: ${{ matrix.database == 'redis' }}
+        name: Start Redis '${{ matrix.version }}'
+        uses: supercharge/redis-github-action@1.2.0
+        with:
+          redis-version: "${{ matrix.version }}"
+
+      - name: Install Task runner
```
Comment (tbuchaillot) [inline .github/workflows/ci-tests.yml:93]: you can use task github action like here https://github.com/TykTechnologies/opentelemetry/blob/main/.github/workflows/ci-tests.yml#L18

File: .github/workflows/ci-tests.yml [hunk 3]
```diff
@@ -108,17 +130,20 @@ jobs:
...
       - name: Install Dependencies
         env:
-          TOKEN: '${{ secrets.ORG_GH_TOKEN }}'
+          TOKEN: "${{ secrets.ORG_GH_TOKEN }}"
         run: >
           git config --global url."https://${TOKEN}@github.com".insteadOf "https://github.com"
 
           go install github.com/wadey/gocovmerge@latest
-      - name: merge reports
+      - name: Install Task runner
```
Comment (tbuchaillot) [inline .github/workflows/ci-tests.yml:141]: is this needed?

File: .github/workflows/ci-tests.yml [hunk 4]
```diff
@@ -73,16 +82,29 @@ jobs:
         name: Start MongoDB '${{ matrix.version }}'
         uses: supercharge/mongodb-github-action@1.8.0
         with:
-          mongodb-version: '${{ matrix.version }}'
-      - name: Run tests
+          mongodb-version: "${{ matrix.version }}"
+
+      - if: ${{ matrix.database == 'redis' }}
```
Comment (tbuchaillot) [inline .github/workflows/ci-tests.yml:87]: What do you think about moving the Redis / mongo initialization to a helper taskfile? 
In that way, development testing can be the same as CI testing.

**Review by tbuchaillot — APPROVED** [review]
Looks good overall @mativm02  - left some small comments

**Pull request merged**